src/module/api/timeline/search/index.ts

```diff
@@ -2,41 +2,38 @@ import apigents from "@venturemark/apigents";
 import * as env from "module/env";
 import * as spec from "./spec";
 
-type DataField = {
-  [key: string]: string;
-};
-
-type ChunkingObject = {
-  perpage: number;
-  pointer: number;
-};
-
-type ApiObject = {
-  chunking: ChunkingObject;
-};
-
-type DataObject = {
-  metadata: DataField;
-};
-
-export interface TimelineSearchI {
-  api?: ApiObject;
-  obj: DataObject[];
-}
-
-export interface TimelineSearchO {
-  api?: ApiObject;
-  obj: DataObject[];
-}
-
-export function Search(request: TimelineSearchI): TimelineSearchO | null {
+export function Search(userId: string, username: string): spec.Res | null {
   {
+    //instantiate classes
     const client = new apigents.Timeline.Client(env.APIEndpoint());
     const req = new apigents.Timeline.Search.I();
 
-    req.setObjList(request);
+    //We don't have definitions in apigents library for SearchI_Obj?
+    const obj = new SearchI_Obj();
```

Comment:
@xh3b4sd the schema requires a SearchI_Obj class input that doesn't exist in "apigents" dst even though it's present in apigents repository typescript definitions.

Reply:
We do only export parts of the generated code we definitely know about. It would require some effort to extend the code generation in a way that it exports secondary types. Can we make this work without specific types? Doesn't this just work with anonymous data structures ala `[{...}]`?

Reply:
I'm not sure - I need to understand proto buffer encoding, serialization better. Let's chat more in sync.

**My current understanding:**
protobuff generates classes which also act as type definitions. Since apigents.Timeline.Search.I is composed of other classes, we need to instantiate those classes and use the setter methods on them to ensure they serialize correctly.

Reply:
My understanding is that Javascript is complete bullshit and you can do pretty much whatever you want. So if there is no supersaiyajin stric mode enabled this should just work with anonymous data structures. The type system in Typescript is guidance for the compiler to generate javascript code. Though this is just a scripting language after all. Maybe just experiment a little bit around. If you are blocked let's get onto a call now and not waste time.

Reply:
Let's chat about this in sync. I believe the error is coming from how the server expects data to be formatted